packages/react-instantsearch-core/src/connectors/connectInfiniteHits.js

```diff
@@ -53,7 +55,10 @@ export default createConnector({
       };
     }
 
-    const { hits, page, nbPages } = results;
+    const { page, nbPages } = results;
+
+    let hits = addAbsolutePositions(results.hits, results.hitsPerPage, page);
```

Comment:
I would rather use two different variables to actually inform that they've changed e.g. `hitsWithPositions`.

Reply:
I agree, let do it!

Reply:
Yes.